### PR TITLE
Add search params (query params) support

### PR DESCRIPTION
**Pull request by skoshy**
This option keeps URL search params in the history and prevents them from being overwritten when react-albus handles navigation. This handles issue #36. 

This feature is very useful when you have a unique link identifier for a user and you want to preserve it in the URL so they can share it with others or return back to it with that data still present in the url.

### CODE REVIEW

**Comment by ldthorne** [comment]
This would be great to have! We built something around this @HiCommon

File: src/components/Wizard.js [hunk 1]
```diff
@@ -82,13 +86,15 @@ class Wizard extends Component {
...
       } else {
-        this.history.replace(`${this.basename}${this.ids[0]}`);
+        this.history.replace(`${this.basename}${this.ids[0]}${this.historySuffix}`);
       }
     });
   };
 
-  push = (step = this.nextStep) => this.history.push(`${this.basename}${step}`);
-  replace = (step = this.nextStep) => this.history.replace(`${this.basename}${step}`);
+  push = (step = this.nextStep) =>
```
Comment (ryhinchey) [inline src/components/Wizard.js:94]: `history.replace` and `history.push` can accept a location-like object which will preserve query params.  check out https://github.com/ReactTraining/history/tree/v4.6.0#navigation.  Using this, you can remove `historySuffix` and do something like this:

```js
replace = (step = this.nextStep) => {
  if (this.props.preserveSearch) {
    this.history.replace({
      ...this.history.location,
      pathname:`${this.basename}${step}`
    })
  } else {
    this.history.replace(`${this.basename}${step}`);
  }
}

```

**Review by bighitbiker3 — APPROVED** [review]

**Comment by teofanis** [comment]
@ryhinchey I could continue if this is needed and @skoshy is ok with it?

**Comment by skoshy** [comment]
Hi! I've lost context on this and am not working with the library recently, so someone else can feel free to take it over and close this PR.

**Comment by Matthew-Mallimo** [comment]
Closing until someone can pick this back up

**Pull request closed without merging**